product-service/src/utils/db/connect.ts

```diff
@@ -0,0 +1,23 @@
+const { Pool, Client } = require('pg');
+
+const pool = new Pool();
+
+let client;
+
+export const createConnection = async (): Promise<typeof Client> => {
+  console.log('Starting initializing db connection');
+
+  client = await pool.connect();
+
+  console.log('Connected to Postgres DB');
+
+  return client;
+};
+
+export const closeConnection = (): void => {
+  console.log('Starting disconnection from db');
+
+  client.release(true);
+
+  console.log('Disconnected from Postgres DB');
+};
```

Comment:
These functions are not pure and unpredictable. What if someone will call client.release and closeConnection in a row? It's better to avoid such side effects

Reply:
It's not pure because of pooling, in this way lambda function is remembering context. But maybe better to implement it with classes, I think about it in future.